File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s = strs[0];
        for(int i=1;i<strs.size();i++){
            int maxLen = min(s.size(),strs[i].size());
            string ans = "";
            for(int j=0;j<maxLen;j++){
                if(s[j]==strs[i][j]) ans +=s[j];
                else break;
            }
            s = ans;
        }
        return s;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int count =0;
        for(int i=0;i<strs[0].length();i++){
            for(int j=1;j<strs.length;j++){
                if(i>=strs[j].length() || strs[j].charAt(i)!=strs[j-1].charAt(i)){ 
                    return strs[0].substring(0,count);
                }   
            }
            count++;
        }
        return strs[0].substring(0,count);
    }
}
